ProcessLib/HeatTransportBHE/BHE/CreateBHECXA.cpp

```diff
@@ -0,0 +1,101 @@
+/**
+ * \file
+ *
+ * \copyright
+ * Copyright (c) 2012-2019, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "CreateBHECXA.h"
+#include "BaseLib/ConfigTree.h"
+#include "CreateFlowAndTemperatureControl.h"
+namespace ProcessLib
+{
+namespace HeatTransportBHE
+{
+namespace BHE
+{
+BHE::BHE_CXA createBHECXA(
+    BaseLib::ConfigTree const& config,
+    std::map<std::string,
+             std::unique_ptr<MathLib::PiecewiseLinearInterpolation>> const&
+        curves)
+{
+    //! \ogs_file_param{prj__processes__process__HEAT_TRANSPORT_BHE__borehole_heat_exchangers__borehole_heat_exchanger__borehole}
+    auto const& borehole_config = config.getConfigSubtree("borehole");
+    const double borehole_length =
+        //! \ogs_file_param{prj__processes__process__HEAT_TRANSPORT_BHE__borehole_heat_exchangers__borehole_heat_exchanger__borehole__length}
+        borehole_config.getConfigParameter<double>("length");
+    const double borehole_diameter =
+        //! \ogs_file_param{prj__processes__process__HEAT_TRANSPORT_BHE__borehole_heat_exchangers__borehole_heat_exchanger__borehole__diameter}
+        borehole_config.getConfigParameter<double>("diameter");
+    BoreholeGeometry const borehole{borehole_length, borehole_diameter};
+
+    //! \ogs_file_param{prj__processes__process__HEAT_TRANSPORT_BHE__borehole_heat_exchangers__borehole_heat_exchanger__pipes}
+    auto const& pipes_config = config.getConfigSubtree("pipes");
+    //! \ogs_file_param{prj__processes__process__HEAT_TRANSPORT_BHE__borehole_heat_exchangers__borehole_heat_exchanger__pipes__outer}
+    Pipe const outer_pipe = createPipe(pipes_config.getConfigSubtree("outer"));
+    Pipe const inner_pipe =
+        //! \ogs_file_param{prj__processes__process__HEAT_TRANSPORT_BHE__borehole_heat_exchangers__borehole_heat_exchanger__pipes__inner}
+        createPipe(pipes_config.getConfigSubtree("inner"));
+    const double pipe_longitudinal_dispersion_length =
+        //! \ogs_file_param{prj__processes__process__HEAT_TRANSPORT_BHE__borehole_heat_exchangers__borehole_heat_exchanger__pipes__longitudinal_dispersion_length}
+        pipes_config.getConfigParameter<double>(
+            "longitudinal_dispersion_length");
+    PipeConfigurationCXA const pipes{inner_pipe, outer_pipe,
+                                     pipe_longitudinal_dispersion_length};
+
+    //! \ogs_file_param{prj__processes__process__HEAT_TRANSPORT_BHE__borehole_heat_exchangers__borehole_heat_exchanger__grout}
+    auto const& grout_config = config.getConfigSubtree("grout");
+    const double grout_density =
+        //! \ogs_file_param{prj__processes__process__HEAT_TRANSPORT_BHE__borehole_heat_exchangers__borehole_heat_exchanger__grout__density}
```

Comment:
Some of the parameter documentations are missing.

Reply:
After changing way of reading parameters, no new parameter is not documented.